File: Frontend/src/contracts/responses/departures/managementDepartureResponse.ts
export interface ManagementDepartureResponse {
    id: number;
    planeModelName: string;
    planePassengerCapacity: number;
    planeImage?: string | null;
    airlineName: string;
    airlineEmail: string;
    airlinePhoneNumber: string;
    airlineImage?: string | null;
    takeOffAirportId: number;
    takeOffAirportName: string;
    takeOffAirportCity?: string | null;
    takeOffAirportIata?: string | null;
    takeOffAirportIcao?: string | null;
    landingAirportId: number;
    landingAirportName: string;
    landingAirportCity?: string | null;
    landingAirportIata?: string | null;
    landingAirportIcao?: string | null;
    requestedTakeOffDateTime: string;
    arrivalDateTime: string;
    createdAt?: string | null;
    submittedAt?: string | null;
    price: number;
    distance: number;
    flightTime: string;
    transfers: number;
    currentStatusId: number;
    statusName: string;
    currentStatusSetAt: string;
    charterRequesterEmail: string;
    charterRequesterFullName?: string | null;
    passengerCount: number;
    canEditRoute: boolean;
    canApprove: boolean;
    canChangeStatus: boolean;
    canDelete: boolean;
    canPay: boolean;
    paymentDeadlineAt?: string | null;
    hasContractDocument: boolean;
    contractDocumentFileName?: string | null;
    contractDocumentUploadedAt?: string | null;
    contractDocumentUploadedByAirline: boolean;
    passengers: ManagementPassengerResponse[];
    employees: ManagementEmployeeResponse[];
    statusHistory: ManagementDepartureStatusResponse[];
    routeAirports: ManagementRouteAirportResponse[];
    routeLegs: ManagementRouteLegResponse[];
}

export interface ManagementPassengerResponse {
    id: number;
    fullName: string;
    email?: string | null;
}

export interface ManagementEmployeeResponse {
    id: number;
    email: string;
    roleName: string;
    fullName?: string | null;
}

export interface ManagementDepartureStatusResponse {
    id: number;
    name: string;
    setAt: string;
}

export interface ManagementRouteAirportResponse {
    id: number;
    name: string;
    city?: string | null;
    country: string;
    iata?: string | null;
    icao?: string | null;
    latitude: number;
    longitude: number;
}

export interface ManagementRouteLegResponse {
    fromAirportId: number;
    toAirportId: number;
    distanceKm: number;
    flightTime: string;
    flightCost: number;
    groundTimeAfterArrival?: string | null;
}

File: AirCharter.API/AirCharter.API/Responses/Departures/ManagementDepartureResponse.cs
using AirCharter.API.Responses.Airports;
using AirCharter.API.Responses.Flights;

namespace AirCharter.API.Responses.Departures
{
    public sealed class ManagementDepartureResponse
    {
        public int Id { get; set; }

        public string PlaneModelName { get; set; } = string.Empty;

        public int PlanePassengerCapacity { get; set; }

        public int TakeOffAirportId { get; set; }

        public string TakeOffAirportName { get; set; } = string.Empty;

        public string? TakeOffAirportCity { get; set; }

        public string? TakeOffAirportIata { get; set; }

        public string? TakeOffAirportIcao { get; set; }

        public int LandingAirportId { get; set; }

        public string LandingAirportName { get; set; } = string.Empty;

        public string? LandingAirportCity { get; set; }

        public string? LandingAirportIata { get; set; }

        public string? LandingAirportIcao { get; set; }

        public DateTime RequestedTakeOffDateTime { get; set; }

        public DateTime ArrivalDateTime { get; set; }

        public DateTime? CreatedAt { get; set; }

        public decimal Price { get; set; }

        public int Distance { get; set; }

        public TimeSpan FlightTime { get; set; }

        public int Transfers { get; set; }

        public int CurrentStatusId { get; set; }

        public string StatusName { get; set; } = string.Empty;

        public DateTime CurrentStatusSetAt { get; set; }

        public string CharterRequesterEmail { get; set; } = string.Empty;

        public int PassengerCount { get; set; }

        public bool CanEditRoute { get; set; }

        public bool CanApprove { get; set; }

        public bool CanChangeStatus { get; set; }

        public IReadOnlyCollection<ManagementPassengerResponse> Passengers { get; set; } =
            Array.Empty<ManagementPassengerResponse>();

        public IReadOnlyCollection<ManagementDepartureStatusResponse> StatusHistory { get; set; } =
            Array.Empty<ManagementDepartureStatusResponse>();

        public IReadOnlyCollection<AirportSearchResponse> RouteAirports { get; set; } =
            Array.Empty<AirportSearchResponse>();

        public IReadOnlyCollection<RouteLegResponse> RouteLegs { get; set; } =
            Array.Empty<RouteLegResponse>();
    }

    public sealed class ManagementPassengerResponse
    {
        public int Id { get; set; }

        public string FullName { get; set; } = string.Empty;

        public string? Email { get; set; }
    }

    public sealed class ManagementDepartureStatusResponse
    {
        public int Id { get; set; }

        public string Name { get; set; } = string.Empty;

        public DateTime SetAt { get; set; }
    }
}
